com.reprezen.swagedit/src/com/reprezen/swagedit/json/references/JsonReferenceValidator.java

```diff
@@ -60,27 +60,25 @@ public JsonReferenceValidator(JsonReferenceFactory factory) {
 	protected Collection<? extends SwaggerError> doValidate(URI baseURI, Iterable<JsonReference> references) {
 		Set<SwaggerError> errors = Sets.newHashSet();
 		for (JsonReference reference: references) {
-			if (!reference.isValid(baseURI)) {
-				Object source = reference.getSource();
-
-				if (source instanceof Node) {
-					errors.add(createReferenceError((Node) source));
-				} else {
-					errors.add(createReferenceError());
-				}
+			if (reference.isInvalid()) {
+				errors.add(createReferenceError(IMarker.SEVERITY_ERROR, reference));
+			} else if (reference.isMissing(baseURI)) {
+				errors.add(createReferenceError(IMarker.SEVERITY_WARNING, reference));
 			}
 		}
 		return errors;
 	}
 
-	protected SwaggerError createReferenceError(Node node) {
-		int line = node.getStartMark().getLine() + 1;
-
-		return new SwaggerError(line, IMarker.SEVERITY_WARNING, Messages.error_invalid_reference);
-	}
+	protected SwaggerError createReferenceError(int severity, JsonReference reference) {
+		Object source = reference.getSource();
+		int line;
+		if (source instanceof Node) {
+			line = ((Node) source).getStartMark().getLine() + 1;
+		} else {
+			line = 1;
+		}
 
-	protected SwaggerError createReferenceError() {
-		return new SwaggerError(1, IMarker.SEVERITY_WARNING, Messages.error_invalid_reference);
+		return new SwaggerError(line, severity, Messages.error_invalid_reference);
```

Comment:
@tedepstein note that we use the same message for warnings (references to inexistent elements) and errors (invalid URLs):
<img width="1027" alt="screen shot 2016-07-14 at 12 04 18 pm" src="https://cloud.githubusercontent.com/assets/644582/16846772/b6828662-49bb-11e6-8a8c-e2e0a53aea3d.png">